packages/types/src/codec/EnumType.ts

```diff
@@ -25,62 +25,83 @@ type Aliasses = {
 };
 
 /**
- * @name EnumType
+ * @name Enum
  * @description
  * This implements an enum, that based on the value wraps a different type. It is effectively
  * an extension to enum where the value type is determined by the actual index.
  */
 // TODO:
 //   - As per Enum, actually use TS enum
 //   - It should rather probably extend Enum instead of copying code
-export default class EnumType<T> extends Base<Codec> implements Codec {
+export default class Enum extends Base<Codec> implements Codec {
   private _def: TypesDef;
   private _index: number;
   private _indexes: Array<number>;
+  private _isBasic: boolean;
 
-  constructor (def: TypesDef, value?: any, index?: number | EnumType<T>, aliasses?: Aliasses) {
-    const decoded = EnumType.decodeEnumType(def, aliasses || {}, value, index);
+  constructor (def: TypesDef | Array<string>, value?: any, index?: number | Enum, aliasses?: Aliasses) {
+    const defInfo = Enum.extractDef(def);
+    const decoded = Enum.decodeEnum(defInfo.def, aliasses || {}, value, index);
 
     super(decoded.value);
 
-    this._def = def;
-    this._indexes = Object.keys(def).map((_, index) => index);
+    this._def = defInfo.def;
+    this._isBasic = defInfo.isBasic;
+    this._indexes = Object.keys(defInfo.def).map((_, index) => index);
     this._index = this._indexes.indexOf(decoded.index) || 0;
   }
 
-  private static decodeEnumType<T> (def: TypesDef, aliasses: Aliasses, value?: any, index?: number | EnumType<T>): Decoded {
+  private static extractDef (def: TypesDef | Array<string>): { def: TypesDef, isBasic: boolean } {
+    if (!Array.isArray(def)) {
```

Comment:
What if I define `new Enum({a: Null, b: Null})`: that should be isBasic. i.e. `isBasic = Object.values(def).every(a=>a === Null)` (can even be a private method instead of a property in this case)

Reply:
A bit sceptical there - since I'm not 100% what the Rust side does. So I know if you do this -

```
enum {
  'Zero',
  'One',
  'Two'
}
```

And it serializes to JSON, it is `1` for `One`. When using types, it actually serializes to the name `One` (or `One: value` when there is one). So it seems slippery - so really keep `isBasic` to just the case where there are strings.